include/envoy/http/filter.h

```diff
@@ -480,6 +489,16 @@ class StreamDecoderFilter : public StreamFilterBase {
    */
   virtual FilterTrailersStatus decodeTrailers(HeaderMap& trailers) PURE;
 
+  /**
+   * Called with decoded metadata. Add new metadata to metadata_map directly. Do not call
+   * StreamDecoderFilterCallbacks::addDecodedMetadata() to add new metadata.
+   *
+   * @param metadata supplies the decoded metadata.
+   */
+  virtual FilterMetadataStatus decodeMetadata(MetadataMap& /* metadata_map */) {
+    return Http::FilterMetadataStatus::Continue;
```

Comment:
Can you add a TODO/substantial comment here about how this default will break filters that do stop and continue style processing and that any use of metadata needs to be carefully audited while we sort out how to handle this generically? (Or something to this effect)

Reply:
Fixed. Thanks!